src/17FEB0.c

```diff
@@ -37,7 +40,477 @@ s32 calc_item_check_hit(void) {
     return HIT_RESULT_HIT;
 }
 
+// Almost matching besides the parts that are marked as "TODO"
+#ifdef NON_MATCHING
+s32 calc_item_damage_enemy(void) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Actor* playerActor = battleStatus->playerActor;
+    s32 currentTargetID = battleStatus->currentTargetID;
+    Actor* partnerActor = battleStatus->partnerActor;
+    s32 currentTargetPartID = battleStatus->currentTargetPart;
+    Actor* targetActor;
+    ActorPart* targetActorPart;
+    ActorState* actorState;
+    Evt* attackScript;
+    s32 sp1C = 0;
+    s32 isFireDamage = 0;
+    s32 isElectricDamage = 0;
+    s32 isIceDamage = 0;
+    s8 wasStatusInflicted = 0;
+    s32 currentHP;
+    s32 baseStatusChance;
+    s32 attackDamage;
+    s32 targetActorDefense;
+    s32 dispatchEvent;
+    u32 sp18;
+    s32 phi_a0_2;
+    s32 phi_fp;
+    
+    battleStatus->wasStatusInflicted = 0;
+    battleStatus->lastAttackDamage = 0;
+    battleStatus->attackerActorID = playerActor->actorID;
+    battleStatus->currentTargetID2 = battleStatus->currentTargetID;
+    battleStatus->currentTargetPart2 = battleStatus->currentTargetPart;
+    
+    targetActor = get_actor(currentTargetID);
+    if (targetActor == NULL) {
+        return 0;
+    }
+
+    targetActorPart = get_actor_part(targetActor, currentTargetPartID);
+    currentTargetID &= 0x700;
+    if (targetActorPart == NULL) {
+        while (1);
```

Comment:
this should be PANIC() and the rest of the function doesn't need to be an else, it can just come after.